### ISSUE TITLE
`test_build_dataset` generates different id's when run within a `test_that` chunk versus a single line

### ISSUE THREAD

**Issue by yangsophieee** [traitecoevo/traits.build#29]
When you run the `test_that` chunk in test-xamples.R, it generates different context id's (e.g. entity_context_id, method_id, etc.) to running each line within the chunk one-by-one. 

I noticed that when I changed the schema file (removed `value_type` from methods table) it also changed the id numbers.

**Comment by yangsophieee** [comment]
I wonder if the `sort` argument in `process_generate_id` would help

**Comment by ehwenk** [comment]
We could sort, but then I think the question is do we want to maintain the original input order or swap to alphabetic order. I think currently in the input order.

**Comment by yangsophieee** [comment]
I agree currently in the input order would be good. I remember that the `sort` argument didn't seem to change anything about the order though when I was looking through your code for `repeat_measurements_id`. This requires further investigation!

**Comment by ehwenk** [comment]
I noted this as well today when trying to rewrite the traits.csv file for `Test_2023_4` and ended up having to manually re-sort and renumber a few rows. It is because in the database being created running outside the test environment, the trait names aren't being rearranged into alphabetical order. It doesn't make sense because `test_build_dataset` seems to have the same process as the normal build process.

**Comment by dfalster** [comment]
Found the cause, and it is more serious than the `test_that` symptom suggests. Fix in #221.

**Why the `test_that` block differs:** `test_that()` pins collation to C via `local_reproducible_output()`. Run the same lines at the console and you get your own locale.

**The actual bug:** ids came from `sort()` and `as.factor()`, both of which collate according to `LC_COLLATE`. So **a build on your machine produces a different database than a build on CI** — the test discrepancy is just how it surfaced.

```
C            NA_2_5  <  female_10_6  <  female_3_3  <  male_1_10
en_AU.UTF-8  female_10_6  <  female_3_3  <  male_1_10  <  NA_2_5
```

`tidyr::unite()` renders `NA` as the literal `"NA"`, so wherever a context has missing values that key moves and every id after it shifts. Building the examples twice changing only `LC_COLLATE`, **four of nine datasets differed**, and the committed `examples/*/output/*.csv` only matched under C.

This was hard to see because `dplyr::arrange()` — the visible ordering in `process.R` — already defaults to C collation and is stable. Only the invisible `sort()` / `as.factor()` steps moved. @ehwenk that is why `test_build_dataset` looked identical to the normal build: it *is* identical, the difference was in the environment, not the code path.

**On the input-order vs alphabetical question:** #221 keeps the current ordering exactly and only makes it locale-independent (`method = "radix"` always collates in C, which is what CI has always produced), so **no existing id changes**. That deliberately sidesteps the ordering preference, which can be settled separately.

**Re `sort` "not changing anything":** it is a no-op for keys that already arrive in sorted order. But there is a real issue underneath — with `sort = FALSE` (the default for `method_id`, `individual_id`, `population_id`, `repeat_measurements_id`) ids depend on row order, which is why editing the schema renumbered them. #221 does not address that; filed as #223 so it survives this closing.

**Comment by dfalster** [comment]
Fixed by #221 (merged into `develop`). `util_sort_locale_independent()` and `util_index_locale_independent()` replace the locale-collated `sort()`/`as.factor()` calls in id generation, and `test-locale.R` builds all nine examples under both `en_US.UTF-8` and `C` and compares them in full.

Since then #229 fixed a second instance of the same class — citations rendered differently depending on `getOption("useFancyQuotes")`, which `test_that()` forces off — so `test-locale.R` now states the general rule in its header: build output must not depend on ambient session settings. Closing.
